tests/rptest/tests/pandaproxy_test.py

```diff
@@ -938,12 +938,30 @@ def _get_brokers(self,
                             headers=headers,
                             auth=auth)
 
+    def _get_topics(self,
+                    headers=HTTP_GET_TOPICS_HEADERS,
+                    auth: Optional[Tuple[str, str]] = None):
+        return requests.get(f"{self._base_uri()}/topics",
+                            headers=headers,
+                            auth=auth)
+
+    def _produce_topic(self,
+                       topic,
+                       data,
+                       headers=HTTP_PRODUCE_BINARY_V2_TOPIC_HEADERS,
+                       auth: Optional[Tuple[str, str]] = None):
+        return requests.post(f"{self._base_uri()}/topics/{topic}",
+                             data,
+                             headers=headers,
+                             auth=auth)
```

Comment:
In general I'm wondering if it would be possible to refactor the existing `PandaProxyTest` tests a little, and perhaps inherit from that to avoid duplication for every endpoint.

Reply:
Of course. This was a draft, I'll add refactored code in the next push.
I usually work in phases:
1) Allow duplicates so test failures are more clear to debug
2) Factor out duplicates with each new push